shared/src/main/java/com/faforever/neroxis/exporter/SCMapExporter.java

```diff
@@ -231,44 +227,34 @@ public static void exportPreview(Path folderPath, SCMap map) throws IOException
-        Path writingPath = folderPath.resolve(decalPath);
+    public static void exportMapwideTexture(Path folderPath, SCMap map) throws IOException {
+        byte[] rawTexture = map.getRawMapTexture();
+        Path textureDirectory = Paths.get("env", "texture");
+        Path filePath = textureDirectory.resolve(MAPWIDE_DDS);
+        Path writingPath = folderPath.resolve(filePath);
         Files.createDirectories(writingPath.getParent());
-        map.getDecals()
-           .add(new Decal(decalParent.resolve(decalPath).toString().replace('\\', '/'), topLeftOffset, new Vector3(),
-                          size, 1000));
         try {
-            Files.write(writingPath, compressedNormal, StandardOpenOption.CREATE);
+            Files.write(writingPath, rawTexture, StandardOpenOption.CREATE);
         } catch (IOException e) {
-            System.out.print("Could not write the normal map image\n" + e);
+            System.out.print("Could not write the map-wide texture\n" + e);
         }
     }
 
-    public static void exportShadows(Path folderPath, SCMap map) throws IOException {
-        float size = map.getPlayableArea().getW() - map.getPlayableArea().getX();
-        Vector2 topLeftOffset = new Vector2(map.getPlayableArea().getX(), map.getPlayableArea().getY());
-        byte[] compressedShadows = map.getCompressedShadows();
-        final String fileFormat = "dds";
-        Path decalsPath = Paths.get("env", "decals");
-        Path decalParent = Paths.get("/maps").resolve(map.getFolderName());
-        Path decalPath = decalsPath.resolve(String.format("map_shadows.%s", fileFormat));
-        Path writingPath = folderPath.resolve(decalPath);
-        Files.createDirectories(writingPath.getParent());
-        Decal shadowDecal = new Decal(decalParent.resolve(decalPath).toString().replace('\\', '/'), topLeftOffset,
-                                      new Vector3(), size, 1000);
-        shadowDecal.setType(DecalType.WATER_ALBEDO);
-        map.getDecals().add(shadowDecal);
-        try {
-            Files.write(writingPath, compressedShadows, StandardOpenOption.CREATE);
-        } catch (IOException e) {
-            System.out.print("Could not write the shadow map image\n" + e);
+    public static void exportPBR(Path folderPath, SCMap map) throws IOException {
+        URL sourceURL = SCMapExporter.class.getResource("/images/heightRoughness.png");
+        if (sourceURL != null) {
+            BufferedImage image = ImageIO.read(sourceURL);
+            Path textureDirectory = Paths.get("env", "texture");
+            Path filePath = textureDirectory.resolve(PBR_DDS);
+            Path writingPath = folderPath.resolve(filePath);
+            Files.createDirectories(writingPath.getParent());
+            try {
+                ImageUtil.writeNormalDDS(image, writingPath);
+            } catch (IOException e) {
+                System.out.print("Could not write the pbr texture\n" + e);
+            }
```

Comment:
Why are we reading it in as an image of we are just copying it? You can just get it as a resourcestream and write that input stream using the nio files API.

Reply:
because I wasn't aware of that solution :D

but I got some post-hoc justifications:
this way it would be easier later to edit the file, as the source file is a png and the file that the map reads should be dds

Reply:
Is it just a raw DDS file with no compression or mip mapping?

Reply:
I think both don't really matter. Compression shouldn't be an issue and the game would generate the mipmaps if they aren't in the file already. I guess having the mipmaps already would improve loading times, but I don't know if the time for that is more than even a fraction of a second